pkg/oidc/provider/token.go

```diff
@@ -408,6 +375,52 @@ func (h *tokenHandler) createTokenResponse(rancherToken *v3.Token, oidcClient *v
 	return resp, nil
 }
 
+func (h *tokenHandler) createIDToken(oidcClient *v3.OIDCClient, rancherToken *v3.Token, scopes []string, user *v3.User, nonce string, groups []string, kid string) *jwt.Token {
+	idClaims := jwt.MapClaims{
+		"aud": []string{oidcClient.Status.ClientID},
+		"exp": h.now().Add(time.Duration(oidcClient.Spec.TokenExpirationSeconds) * time.Second).Unix(),
+		"iss": settings.ServerURL.Get() + "/oidc",
+		"iat": h.now().Unix(),
+		"sub": rancherToken.UserID,
+	}
+
+	if slices.Contains(scopes, "profile") {
+		idClaims["name"] = user.DisplayName
+	}
+	if nonce != "" {
+		idClaims["nonce"] = nonce
+	}
+	if groups != nil {
+		idClaims["groups"] = groups
+	}
+	if rancherToken.AuthProvider != "" {
+		idClaims["auth_provider"] = rancherToken.AuthProvider
+	}
+	idToken := jwt.NewWithClaims(jwt.SigningMethodRS256, idClaims)
+	idToken.Header["kid"] = kid
+	return idToken
+}
+
+// CreateAccessToken creates and returns a JWT access token.
+func CreateAccessToken(oidcClient *v3.OIDCClient, rancherToken *v3.Token, scopes []string, kid string, now time.Time) *jwt.Token {
+	accessClaims := jwt.MapClaims{
+		"aud":   []string{oidcClient.Status.ClientID},
+		"exp":   now.Add(time.Duration(oidcClient.Spec.TokenExpirationSeconds) * time.Second).Unix(),
+		"iss":   settings.ServerURL.Get() + "/oidc",
+		"iat":   now.Unix(),
+		"sub":   rancherToken.UserID,
+		"scope": scopes,
+		"token": rancherToken.Name + ":" + rancherToken.Token,
```

Comment:
we might want to consider adding only the `rancherToken.Name` here as the `rancherToken` probably expires after the `access_token`. However, that would mean and extra call to the cache to get the token value when authorizing. What do you think?